api/src/helpers/week.ts

```diff
@@ -49,24 +61,59 @@ export function getWeek(): Promise<WeekData> {
  * @param quarterMapping Maps a quarter to its start and end date 
  * @returns Week description if it lies within the quarter
  */
-function findWeek(date: Date, quarterMapping: QuarterMapping): WeekData {
+function findWeek(date: dayjs.Dayjs, quarterMapping: QuarterMapping): WeekData {
     let result: WeekData = undefined!;
     // iterate through each quarter
     Object.keys(quarterMapping).forEach(function (quarter) {
-        let begin = new Date(quarterMapping[quarter]['begin'])
-        let end = new Date(quarterMapping[quarter]['end'])
+        let beginDate = new Date(quarterMapping[quarter]['begin']);
+        let endDate = new Date(quarterMapping[quarter]['end']);
+
+        let begin: dayjs.Dayjs;
+        let end: dayjs.Dayjs;
+
+        // begin/end dates are incorrectly in UTC+0, likely due to AWS servers being in UTC+0 by default
+        // so for example, Winter 2023 starts on Monday Jan 9, 2023 PST
+        // the begin date on production is incorrectly Jan 9, 2023 00:00 UTC+0, when it should be
+        // Jan 9, 2023 0:00 UTC-8 (since Irvine is in PST which is 8 hours behind UTC)
+        // we want to fix this offset for accurate comparsions
+        if (beginDate.getUTCHours() === 0) {
+            begin = dayjs.tz({ year: beginDate.getUTCFullYear(), month: beginDate.getUTCMonth(), date: beginDate.getUTCDate() });
+            end = dayjs.tz({ year: endDate.getUTCFullYear(), month: endDate.getUTCMonth(), date: endDate.getUTCDate() });
+        } else { // default case if the dates aren't in UTC+0 and are in correct timezone
+            begin = dayjs(beginDate).tz();
+            end = dayjs(endDate).tz();
+        }
+
+        // adjust instruction end date to last ms of the day
+        end = end.add({
+            hours: 23,
+            minutes: 59,
+            seconds: 59,
+            ms: 999
+        });
+
+        let isFallQuarter = false;
+        // in fall quarter, instruction start date is not on a monday
+        // it is on a thursday in week 0
+        // let's move it back to monday in week 0 and adjust our week calculations by -1
+        // that way week calculations are correct for fall quarter
+        if (begin.day() !== 1) {
```

Comment:
It would be good to set a single ENUM called Monday == 1
https://www.typescriptlang.org/docs/handbook/enums.html